edgedb/_testbase.py

```diff
@@ -52,52 +52,58 @@ def flt(log_record):
 _default_cluster = None
 
 
-def _init_cluster(data_dir=None, *, cleanup_atexit=True):
-    if (not os.environ.get('EDGEDB_DEBUG_SERVER') and
-            not os.environ.get('EDGEDB_LOG_LEVEL')):
-        _env = {'EDGEDB_LOG_LEVEL': 'silent'}
-    else:
-        _env = {}
-
-    if data_dir is None:
-        cluster = edgedb_cluster.TempCluster(env=_env, testmode=True)
-        destroy = True
-    else:
-        cluster = edgedb_cluster.Cluster(data_dir=data_dir, env=_env)
-        destroy = False
-
-    if cluster.get_status() == 'not-initialized':
-        cluster.init()
-
-    cluster.start(port='dynamic')
-    cluster.set_superuser_password('test')
-
-    if cleanup_atexit:
-        atexit.register(_shutdown_cluster, cluster, destroy=destroy)
-
-    return cluster
-
-
 def _start_cluster(*, cleanup_atexit=True):
     global _default_cluster
 
-    if _default_cluster is None:
-        cluster_addr = os.environ.get('EDGEDB_TEST_CLUSTER_ADDR')
-        if cluster_addr:
-            conn_spec = json.loads(cluster_addr)
-            _default_cluster = edgedb_cluster.RunningCluster(**conn_spec)
-        else:
-            data_dir = os.environ.get('EDGEDB_TEST_DATA_DIR')
-            _default_cluster = _init_cluster(
-                data_dir=data_dir, cleanup_atexit=cleanup_atexit)
+    if isinstance(_default_cluster, Exception):
+        # when starting a server fails
+        # don't retry starting one for every TestCase
+        # because that takes a long time
+        raise _default_cluster
+
+    if _default_cluster:
+        return _default_cluster
+
+    args = [
+        os.environ.get('EDGEDB_SERVER_BINARY', 'edgedb-server'),
+        "--temp-dir",
+        "--testmode",
+        "--echo-runtime-info",
```

Comment:
Use `--emit-server-status=` instead.  That way you'd be able to leave stdin/stderr undisturbed, which, in turn, will simplify debug.

Reply:
changed